sanity-io/structured-content-2022
# External links in new tab / new link structure

# External links in new tab

## Intent

Open rich text "External link"s in a new tab if the "Open in new tab" setting has been switched on, and similarly for richtext "Link" objects and for "Simple call to action" objects.

## Description

This has been discussed in the Shortcut story "[Minor design implementation tweaks (first round)](https://app.shortcut.com/sanity-io/story/16013/minor-design-implementation-tweaks-first-round)", which includes this task ("Open new tab when clickling on external links"). This PR doesn't open _all_ external links in new tabs, just some of them. Mainly, those richtext or `simpleCallToAction` links where the "Open in new tab or window" setting has been enabled.

The PR also currently sets the "Tickets" links in header nav and "nav block" to open in a new tab, because it was requested by Sindre in the story. Would be good to have confirmation from @kmelve if this is OK or if it should be reverted. (As discussed in the story there are also other links that already open in new tabs prior to this PR, these are in the footer and cookie banner.)

## Testing this PR

1. Ensure you're using a browser config with the usual behavior of defaulting to opening links in the current tab (with the possible exception of downloads, non-HTTP/HTTPS URI schemes etc)
2. Open https://structured-content-2022-web-git-external-links-in-new-tab.sanity.build/link-testing
3. Verify that all links/button on the page are working (the ones marked "internal" point to the /programs page, and the ones marked "external" point to sanity.io) and that the ones marked "new tab" open in a new tab (or a new window if your browser is configured like that)
